src/odin/ingestion/afc/afc_restricted.py

```diff
@@ -0,0 +1,100 @@
+import os
+import sched
+from pathlib import Path
+
+
+from odin.job import job_proc_schedule
+from odin.utils.runtime import sigterm_check
+from odin.utils.logger import ProcessLog
+from odin.utils.locations import DATA_SPRINGBOARD
+from odin.utils.locations import AFC_RESTRICTED
+from odin.utils.aws.s3 import list_objects
+from odin.utils.aws.s3 import download_object
+from odin.utils.aws.s3 import upload_file
+from odin.utils.parquet import ds_from_path
+from odin.utils.parquet import pq_dataset_writer
+
+import polars as pl
+from odin.ingestion.afc.afc_archive import ArchiveAFCAPI
+
+
+class RestrictedAFC(ArchiveAFCAPI):
+    """Custom class for AFC Restricted Table"""
+
+    def sync_parquet(self) -> None:
+        """Convert json to parquet and sync with S3 files."""
+        log = ProcessLog("afc_api_sync_parquet")
+        sync_paths = []
+        rs_sync_paths = []
+        for json_file in sorted(Path(self.tmpdir).iterdir(), key=os.path.getmtime):
+            if json_file.suffix != ".json":
+                continue
+            lf = pl.scan_ndjson(
+                json_file.absolute(),
+                schema=self.schema,
+            ).with_columns(pl.col(self.ts_cols).str.head(19).str.to_datetime("%Y-%m-%d %H:%M:%S"))
+
+            # create public parquet file
+            pq_path = str(json_file.absolute()).replace(".json", ".parquet")
+            lf.drop("engraveid").sink_parquet(
+                pq_path,
+                compression="zstd",
+                compression_level=3,
+                row_group_size=int(1024 * 1024 / (8 * self.schema.len())),
+            )
+
+            # create restricted lookup file
+            rs_pq_path = str(json_file.absolute()).replace(".json", "_restricted.parquet")
+            lf.select(["mediumid", "engraveid"]).unique().sink_parquet(
+                rs_pq_path,
+                compression="zstd",
+                compression_level=3,
+                row_group_size=int(1024 * 1024 / (8 * self.schema.len())),
+            )
+
+            json_file.unlink()
+            pq_row_count = ds_from_path(pq_path).count_rows()
+            if pq_row_count > 0:
+                sync_paths.append(pq_path)
+                rs_sync_paths.append(rs_pq_path)
+                log.add_metadata(pq_path=pq_path, pq_row_count=pq_row_count)
+
+        if len(sync_paths) == 0:
+            return
+
+        export_jobs = (
+            (self.export_folder, sync_paths),
+            (os.path.join(DATA_SPRINGBOARD, AFC_RESTRICTED, self.table), rs_sync_paths),
+        )
+        new_paths: list[str] = []
```

Comment:
Run same "merge" process for both datasets, only difference is "export_path" and source "sync_paths)